### ISSUE TITLE
strictNullCheck False Posivitve when access propery on created object.

### ISSUE THREAD

**Issue by Bene-Graham** [Microsoft/TypeScript#31202]
**TypeScript Version:**  3.5.0-dev.20190501

**Search Terms:**
strict false positive object

**Code**

```ts
import * as azure from "@azure/storage-blob";

interface ISaveBlobOptions {
    blobName: string;
    contentDisposition?: {
        fileName: string;
        type: "inline" | "attachment";
    };

    contentType: string;

    fileData: Buffer;
}

const saveBlobBase = async (input: ISaveBlobOptions): Promise<void> => {
    this.validateSaveBlobBase(input);

    await this.createStorage();

    const options: azure.IBlockBlobUploadOptions = {
        blobHTTPHeaders: {
            blobContentType: input.contentType
        }
    };

    if(input.contentDisposition) {
        // Getting Object is possibly 'undefined'.
        options.blobHTTPHeaders.blobContentDisposition = `${input.contentDisposition.type}; filename="${input.contentDisposition.fileName}"`;
    }

    if (input.fileData instanceof Buffer) {
        await this.saveBlobBuffer(input.blobName, input.fileData, options);
    } else if(this.isMulterFile(input.fileData)) {
        await this.saveBlobMulterFile(input.blobName, input.fileData, options);
    } else {
        throw new Error("Unknown type for param file");
    }
}

saveBlobBase({
    blobName: "myBlobName",
    contentDisposition: {
        fileName: "foo.bar",
        type: "inline"
    },
    contentType: "text/plain",
    fileData: Buffer.from("SOME TEXT")
});
```

tsconfig
```json
{
	"compilerOptions": {
		"target": "es2017",
		"module": "commonjs",
		"sourceMap": true,
		"strictPropertyInitialization": true,
		"strictNullChecks": true
	}
}
```

**Expected behavior:**

since when the creating the ```options``` variable the property ```blobHTTPHeaders``` is also defined. I would expect that accessing properties off of ```blobHTTPHeaders``` should not throw a "Object is possibly 'undefined'."

**Actual behavior:**

Getting Object is possibly 'undefined'.

**Playground Link:** 

**Related Issues:**

**Comment by AlCalzone** [comment]
According to https://docs.microsoft.com/de-de/javascript/api/@azure/storage-blob/iblockblobuploadoptions?view=azure-node-preview `blobHTTPHeaders` is an optional property on `azure.IBlockBlobUploadOptions`.

You told TypeScript that `options` is of type `azure.IBlockBlobUploadOptions`, which means that `options.blobHTTPHeaders` is optional. Whether you assign an object to it that has `blobHTTPHeaders` doesn't matter in that case. `options.blobHTTPHeaders` is not narrowed to "not undefined" in your code.

You can solve the problem by explicitly telling TypeScript how `options` looks like - in your case it is a `{ blobHTTPHeaders: BlobHTTPHeaders}`.

**Comment by AlCalzone** [comment]
On 2nd thought, this seems to be a duplicate of https://github.com/Microsoft/TypeScript/issues/20219
